### PR TITLE
Flake8 and integration tests

### PR DESCRIPTION
**Pull request by sbesson**
Following the initial roadmap discussed in https://github.com/ome/omero-metadata/pull/1#issuecomment-385521809, this PR mostly activates the remaining integration tests in preparation of new features opened against this decoupled plugins

Summary of the changes:

- consumes the `cli-build` from https://github.com/openmicroscopy/omero-test-infra/pull/21 to execute the integration tests not located under `test/integration/clitest` during the Travis CI build 
- restores the missing YML/CSV test files from the initial filter-branching. Ownership of the initial commits (c58d31d...ca5ef2e) is preserved but can be squashed if desired.
- fixes the source code and tests to be flake8 compliant and comply with the upstream changes proposed in https://github.com/openmicroscopy/omero-test-infra/pull/21 

--depends-on https://github.com/openmicroscopy/omero-test-infra/pull/21

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ services:
   - docker
 
 before_install:
-  - git clone  git://github.com/openmicroscopy/omero-test-infra .omero
+  - git clone --recurse-submodules git://github.com/sbesson/omero-test-infra -b cli_pycheck .omero
```
Comment (joshmoore) [inline .travis.yml]: Why are you recursing? Just in case?

Comment (sbesson) [reply .travis.yml]: I think brute copy-n-paste, I plan to revert this commit once we have the upstream omero-test-infra PR integrated

**Comment by jburel** [comment]
@joshmoore's suggestion makes sense. i.e. moving to test-infra
Any project can override the scope of an action
This is already the case

**Comment by joshmoore** [comment]
> restores the missing YML/CSV test files from the initial filter-branching

:+1: for the original ownership of the changes, but can you refresh our collective (or at least my faulty) memory of where this puts us in the tagging regime?

Otherwise, changes seems fine.

**Comment by sbesson** [comment]
Given the nature of the changes, proposing to get this released as `0.2.2`. Beyond this tag, I do not know of any other priority feature on our side.
So I think having the plugin decoupled/released with the code made compliant and all existing integration tests activated and tested via Travis CI is a strong milestone to finish this phase i.e. `0.2.x`. From there, anyone with concrete driver should feel free to initiate the work on the next round of features i.e. move towards `0.3.0`.

**Comment by joshmoore** [comment]
Summarizing after looking at https://github.com/openmicroscopy/openmicroscopy/pull/5691#issuecomment-409603202 so `0.2.x` intends to represent 5.4.7, and 0.2.2 is another adjustment to bring them closer?

**Comment by sbesson** [comment]
That's correct. Apart from the new CLI wrapper `src/omero/plugins/metadata.py`, the complete diff between the source code as in OMERO 5.4.7 and `omero-metadata 0.2.2` is:

```diff
sbesson@ls30630:omero-metadata (master) $ diff -ru OMERO.py-5.4.7-ice36-b93/lib/python/omero/util/populate_metadata.py src/populate_metadata.py 
--- OMERO.py-5.4.7-ice36-b93/lib/python/omero/util/populate_metadata.py	2018-07-25 09:48:00.000000000 +0100
+++ src/populate_metadata.py	2018-08-15 15:21:43.000000000 +0100
@@ -54,7 +54,7 @@
 from omero.util import pydict_text_io
 from omero import client
 
-from populate_roi import ThreadPool
+from omero.util.populate_roi import ThreadPool
 
 
 log = logging.getLogger("omero.util.populate_metadata")
@@ -87,6 +87,7 @@
 Report bugs to ome-devel@lists.openmicroscopy.org.uk""" % (error, cmd, cmd)
     sys.exit(2)
 
+
 # Global thread pool for use by workers
 thread_pool = None
 
@@ -1655,10 +1656,10 @@
                                        loops, ms)
 
     def _write_to_omero_batch(self, to_delete, loops=10, ms=500):
-        delCmd = omero.cmd.Delete2(targetObjects=to_delete)
+        del_cmd = omero.cmd.Delete2(targetObjects=to_delete)
         try:
             callback = self.client.submit(
-                delCmd, loops=loops, ms=ms, failontimeout=True)
+                del_cmd, loops=loops, ms=ms, failontimeout=True)
         except CmdError, ce:
             log.error("Failed to delete: %s" % to_delete)
             raise Exception(ce.err)
sbesson@ls30630:omero-metadata (master) $ diff -ru OMERO.py-5.4.7-ice36-b93/lib/python/omero/plugins/metadata.py src/omero_cli_metadata.py 
--- OMERO.py-5.4.7-ice36-b93/lib/python/omero/plugins/metadata.py	2018-07-25 09:48:00.000000000 +0100
+++ src/omero_cli_metadata.py	2018-08-15 15:21:43.000000000 +0100
@@ -12,15 +12,15 @@
 import mimetypes
 import os
 import re
-import sys
 
 import omero
 from omero.cli import BaseControl
-from omero.cli import CLI
+
 from omero.cli import ProxyStringType
 from omero.constants import namespaces
 from omero.gateway import BlitzGateway
-from omero.util import populate_metadata, populate_roi, pydict_text_io
+import populate_metadata
+from omero.util import populate_roi, pydict_text_io
 from omero.util.metadata_utils import NSBULKANNOTATIONSCONFIG
 from omero.util.metadata_utils import NSBULKANNOTATIONSRAW
 from omero.grid import LongColumn
@@ -423,7 +423,7 @@
         try:
             table.initialize([LongColumn('ID', '', [])])
             initialized = True
-        except:
+        except Exception:
             pass
         finally:
             table.close()
@@ -431,7 +431,7 @@
         try:
             orig_file = table.getOriginalFile()
             conn.deleteObject(orig_file)
-        except:
+        except Exception:
             # Anything else to do here?
             pass
 
@@ -621,16 +621,6 @@
             if args.z:
                 pixel.setPhysicalSizeZ(omero.model.LengthI(args.z, unit))
 
-        groupId = pixels[0].getDetails().getGroup().getId().getValue()
-        ctx = {'omero.group': str(groupId)}
+        group_id = pixels[0].getDetails().getGroup().getId().getValue()
+        ctx = {'omero.group': str(group_id)}
         conn.getUpdateService().saveArray(pixels, ctx)
-
-
-try:
-    if "OMERO_DEV_PLUGINS" in os.environ:
-        register("metadata", MetadataControl, HELP)
-except NameError:
-    if __name__ == "__main__":
-        cli = CLI()
-        cli.register("metadata", MetadataControl, HELP)
-        cli.invoke(sys.argv[1:])
```

i.e. mostly adjustements to the decoupling and flake8 compliant changes.

**Pull request merged**